taigit/src/libraries/Taiga.ts

```diff
@@ -78,4 +78,124 @@ task_history(taskId : number) : Promise<Object> {
+ * * {
+ * Current in Use
+ *      state_trans_invalid: Boolean       // true, the task state transition is not valid; false, the task is valid.
+ *      date : number,         // Date and time of history entry in milliseconds since epoch
+ *      user : object,         // Taiga User Object
+ *      status_trans:entry.String[],//Status transition array
+ *
+ *  Can be added
+ *      diff_types : String[], // Names of entries in diff_values //diff_values : entry.values_diff,
+ *      diff_values : Object   // Values of the changes listed in diff_types //diff_values : entry.values_diff,
+ * }
+ */
+export async function
+task_assessment_state_trans(taskId : number) : Promise<Object> {
+let data = (await axios.get(`https://api.taiga.io/api/v1/history/task/${taskId}`)).data;
+    //test case :https://api.taiga.io/api/v1/history/task/2577741
+let output : Array<Object> = [];
+for(let entry of data) {
+    let new_entry = {
+        state_trans_invalid: false,
+        date : new Date(entry.created_at).getTime(),
+        user : entry.user,
+        status_trans:entry.values_diff.status
+    }
+
+    //Only three status transistion is valid
+    //which is  ["New", "In progress"] ["In progress", "Ready for test"]  ["Ready for test", "Closed"])
+    if((new_entry.status_trans[0]!="New"&&new_entry.status_trans[1]!="In progress")
+        &&(new_entry.status_trans[0]!="In progress"&&new_entry.status_trans[1]!="Ready for test")
+        &&(new_entry.status_trans[0]!="Ready for test"&&new_entry.status_trans[1]!="Closed"))
+        new_entry.state_trans_invalid= true;
+    output.push(new_entry);
+}
+
+return output;
+}
+
+
+**
+ * @summary This call return a task current end status and estimated finished percentage based on task Id
+ * @param taskId the ID for the task to assess task
+ * @returns  whether the task is abnormal,task status transition, date based on task Id
+ * * {
+ * Current in Use
+ *      finished: Boolean           // true, the task is finished.
+ *      fin_per: number            // finished percentage % default [0%,10%,60%,100%]
+ *      end_status: string,       // current task status  ["New", "In progress","Ready for test", "Closed"]
+ *      num_stat:number           // current task status [0 1 2 3] corresponding to string in end_status
+ *
+ * }
+ */
+export async function
+task_assessment_endstate(taskId : number) : Promise<Object> {
+
+    let data = (await axios.get(`https://api.taiga.io/api/v1/history/task/${taskId}`)).data;
+    //test case :https://api.taiga.io/api/v1/history/task/2577741   or 2555550
+    let output : Array<Object> = [];
+    let endstatus:string = "New";
+    for(let entry of data) {
+        let new_entry = {
+            state_trans_invalid: false,
+            date : new Date(entry.created_at).getTime(),
+            user : entry.user,
+            status_trans:entry.values_diff.status
+        }
+        endstatus=new_entry.status_trans[1];
```

Comment:
static analysis: you have inconsistent variable name styles. please pick a style (or review previous methods to see what the trend is) and stick with it. If there is inconsistency across all functions please alert me of this so I can address this in my documentation task for earlier functions

Reply:
fixed.